frontend/components/CourseActions/components/CourseDescriptionAndStats.js

```diff
@@ -12,6 +14,31 @@ class CourseDescriptionAndStats extends React.Component {
     My: PropTypes.object.isRequired
   }
 
+  state = {
+    stats: []
+  }
+
+  componentDidMount() {
+    this.getStats();
+  }
+
+  getStats = () =>
+    api.CourseApi.getStudentsStats(
+      (spe) => spe.status === 'success',
+      { courseId: this.props.courseDto.course.id, authorId: this.props.courseDto.author.id }
+    )
+    .then((payload) => {
+      this.setState({ stats: [] })
+      const merged_learners = []
+
+      for (let index = 0; index < payload.length; index++) {
+        const element = payload[index];
+        const merged = { ...this.props.courseDto.learners[index], ...element };
+        merged_learners.push(merged)
+        this.setState({ stats: merged_learners })
```

Comment:
Hm I see the logic, you're merging `this.props.courseDto.learners` we already fetched with the new data.
I suggest we ignore the `this.props.courseDto` data structure (now that we have real stats I'll make the `/api/PageApi.getForCourseActions` api call simply return the number of students and nothing more, so there won't be any students to merge the info with), this will simplify the code a great deal.

Reply:
Ok, I got rid of that merge logic and simply used the stats object to show things in the modal. Much straight forward now.